mobile_app_connector/models/firebase_notification.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    @author: Nicolas Badoux <n.badoux@hotmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import api, fields, models
+from odoo.fields import Datetime
+from datetime import datetime
+
+
+class FirebaseNotification(models.Model):
+    _inherit = "firebase.notification"
+
+    destination = fields.Selection([
+        ('MyHub', 'MyHub'),
+        ('Letter', 'Letter'),
+        ('Donation', 'Donation'),
+        ('Prayer', 'Prayer'),
+    ], default="MyHub")
+
+    topic = fields.Selection([
+        ('general_notification', 'General channel'),
+        ('child_notification', 'Child channel'),
+        ('spam', "None (bypass user's preferences)")
+    ], default='general_notification', required=True)
+
+    fundType = fields.Many2one('product.product')
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+
+    @api.multi
+    def send(self, data=None):
+        """
+        Filters notifications w.r.t. user's preference
+
+        :param data:
+        :return:
+        """
+        for notif in self:
+            data = {
+                'topic': notif.topic,
+                "destination": notif.destination or "",
+                "fund_type_id": str(notif.fundType.id)
+            }
+            super(FirebaseNotification, self).send(data)
```

Comment:
```suggestion
            super(FirebaseNotification, notif).send(data)
```
Otherwise you will call the super with all recordsets at each iteration.